src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/features/form/validation/validationSlice.ts

```diff
@@ -0,0 +1,91 @@
+import { createAction, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { IComponentValidations, IValidations } from 'src/types';
+
+export interface IValidationState {
+  validations: IValidations;
+  invalidDataTypes: string[];
+  error: Error;
+  currentSingleFieldValidation: string;
+}
+
+export interface IUpdateComponentValidations {
+  layoutId: string;
+  validations: IComponentValidations;
+  componentId: string;
+  invalidDataTypes?: string[];
+}
+
+export interface IUpdateValidations {
+  validations: IValidations;
+}
+
+export interface IValidationActionRejected {
+  error: Error;
+}
+
+export interface ISetCurrentSingleFieldValidationAction {
+  dataModelBinding?: string;
+}
+
+const initialState: IValidationState = {
+  validations: {},
+  error: null,
+  invalidDataTypes: [],
+  currentSingleFieldValidation: null,
+};
+
+const moduleName = 'validation';
+
+const validationSlice = createSlice({
+  name: moduleName,
+  initialState,
+  reducers: {
```

Comment:
As mentioned earlier I assume the following block is the main reason for the disabled lint rule.
Found this in the Redux Toolkit docs:
https://redux-toolkit.js.org/usage/immer-reducers#linting-state-mutations